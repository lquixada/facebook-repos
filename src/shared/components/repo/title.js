import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

export const Title = ({children}) => (
  <Wrapper>{children}</Wrapper>
);

Title.propTypes = {
  children: PropTypes.node,
};

const Wrapper = styled.h2`
  color: #3c4146;
  font-family: 'Lato', Helvetica, sans-serif;
  font-size: 3rem;
  font-weight: normal;
  letter-spacing: -.1rem;
  line-height: 1.5;
  vertical-align: middle;
`;
