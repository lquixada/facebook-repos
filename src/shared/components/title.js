import React from 'react'
import {Link} from 'react-router-dom'
import styled from 'styled-components'

export const Title = () => (
  <Wrapper>
    <NavLink to='/'>
      <Image src='/assets/images/logo.jpg' />
    </NavLink>
  </Wrapper>
)

const Wrapper = styled.h1`
  line-height: 1;
`

const NavLink = styled(Link)`
  display: inline-block;
  height: 4.5rem;
  font-size: 1rem;
  text-decoration: none;
`

const Image = styled.img`
  display: inline-block;
  height: 100%;
`
