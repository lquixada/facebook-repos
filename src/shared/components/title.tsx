import React from "react";
import {Link} from "react-router-dom";
import styled from "styled-components";

import {imgUrl} from "../helpers";

export const Title = () => (
  <Wrapper>
    <NavLink to="/">
      <Image src={imgUrl("logo.png")} />
    </NavLink>
  </Wrapper>
);

const Wrapper = styled.h1`
  line-height: 1;

  &::after {
    content: "/";
    margin: 0 5px;
    color: #b9b9b9;
    font-family: 'Lato', Helvetica, sans-serif;
    font-size: 25px;
    font-weight: normal;
    vertical-align: middle;
  }
`;

const NavLink = styled(Link).attrs({
  "aria-label": "Homepage",
})`
  display: inline-block;
  height: 30px;
  vertical-align: middle;
  text-decoration: none;
`;

const Image = styled.img.attrs({
  alt: "Github Repos logo",
})`
  display: inline-block;
  width: 158px;
  height: 100%;
`;
