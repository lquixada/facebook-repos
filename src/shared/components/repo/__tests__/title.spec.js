import React from 'react';
import {mount} from 'enzyme';

import {Title} from '../title';

describe('<Title />', () => {
  it('matches the snapshot', () => {
    const component = mount(<Title>Facebook Repos</Title>);
    expect(component).toMatchSnapshot();
  });
});
