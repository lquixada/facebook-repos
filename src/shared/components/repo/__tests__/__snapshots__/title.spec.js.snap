// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> matches the snapshot 1`] = `
<Title>
  <styled.h2>
    <h2
      className="sc-bdVaJa ketyGn"
    >
      Facebook Repos
    </h2>
  </styled.h2>
</Title>
`;
